Library/Homebrew/keg.rb

```diff
@@ -295,14 +295,24 @@ def lock
   end
 
   def completion_installed?(shell)
-    dir = case shell
-    when :bash then path.join("etc", "bash_completion.d")
-    when :zsh  then path.join("share", "zsh", "site-functions")
-    when :fish then path.join("share", "fish", "vendor_completions.d")
+    case shell
+    when :bash
+      dir = path.join("etc", "bash_completion.d")
+    when :zsh
+      # By convention, zsh completion files start with an underscore
+      return !Dir["#{path}/share/zsh/site-functions/_*"].empty?
+    when :fish
+      dir = path.join("share", "fish", "vendor_completions.d")
     end
     dir && dir.directory? && !dir.children.empty?
   end
 
+  def zsh_functions_installed?
+    # Check for non completion functions (i.e. files not started with an underscore),
+    # since those can be checked separately
+    !Dir["#{path}/share/zsh/site-functions/*"].reject{ |f| f[0] == "_" }.empty?
```

Comment:
Thanks!
Do y'all want me to amend this commit with changes or create new commits?